### ISSUE TITLE
[pickers] `disableOpenPicker` prop not working on MobileDatePicker

### ISSUE THREAD

**Issue by julianCast** [mui/mui-x#8036]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example on v5: https://codesandbox.io/s/heuristic-solomon-mxsrc0?file=/demo.tsx
Link to live example on v6: https://codesandbox.io/s/distracted-tom-6m02r8?file=/demo.tsx

I used the example of your documentation, adding the `disableOpenPicker` to the three different date pickers. Only MobileDatePicker will still be able to be opened.

### Current behavior 😯

MobileDatePicker ignores the `disableOpenPicker` and it's opened with interaction.

### Expected behavior 🤔

MobileDatePicker should not open the picker after `disableOpenPicker` is used.

### Your environment 🌎
 This is my environment but, as I said, it's happening in your documentation example.
<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
 System:
    OS: macOS 13.0.1
  Binaries:
    Node: 16.17.1 - ~/.nvm/versions/node/v16.17.1/bin/node
    Yarn: 3.3.0 - ~/.nvm/versions/node/v16.17.1/bin/yarn
    npm: 8.15.0 - ~/.nvm/versions/node/v16.17.1/bin/npm
  Browsers:
    Chrome: 110.0.5481.100
    Edge: 110.0.1587.41
    Firefox: 108.0.1
    Safari: 16.1
  npmPackages:
    @emotion/react: ^11.10.5 => 11.10.5
    @emotion/styled: ^11.10.5 => 11.10.5
    @mui/base:  5.0.0-alpha.117
    @mui/core-downloads-tracker:  5.11.8
    @mui/lab: ^5.0.0-alpha.119 => 5.0.0-alpha.119
    @mui/material: ^5.11.8 => 5.11.8
    @mui/private-theming:  5.11.7
    @mui/styled-engine:  5.11.8
    @mui/system:  5.11.8
    @mui/types:  7.2.3
    @mui/utils:  5.11.7
    @mui/x-date-pickers: ^5.0.15 => 5.0.15
    @types/react: ^17.0.39 => 17.0.39
    react: 17.0.2 => 17.0.2
    react-dom: 17.0.2 => 17.0.2
    typescript: ^4.9.4 => 4.9.4
```
</details>

**Comment by LukasTy** [comment]
[API documents](https://mui.com/x/api/date-pickers/date-picker/#:~:text=for%20your%20format.-,disableOpenPicker,-bool) this prop and mentions that it controls the rendering of the `open picker button`.
Is there a use case you are trying to achieve and this behavior is stopping you from doing it?
IMO we should be making some sort of a change in this regard. Either:
- change the description and also not open view on mobile when it is toggled on;
- remove this prop from available options on `Mobile` picker versions;

What's your take @flaviendelangle?

**Comment by julianCast** [comment]
Oh I misunderstood the meaning of it. I was trying to disable the opening. I think the first option would make sense :

> change the description and also not open view on mobile when it is toggled on;

Thank you for the clarification!

**Comment by LukasTy** [comment]
> Oh I misunderstood the meaning of it. I was trying to disable the opening. I think the first option would make sense :

After discussing and investigating this issue further, it's clear that simply disabling the mobile picker opening would sort of make no sense, because that is the only behavior that is currently supported by the mobile picker versions.

This change should be done together with or after #7869.

**Comment by michelengelen** [comment]
Thanks for the report and the discussion. We resolved this in the opposite direction. The `disableOpenPicker` prop was deprecated in [v8.0.0](https://github.com/mui/mui-x/releases/tag/v8.0.0) (#17040) and will be removed in the next major. For your confirmed use case (editing only through the input, no picker opening), use the standalone [field components](https://mui.com/x/react-date-pickers/fields/) instead:

```tsx
-<MobileDatePicker disableOpenPicker />
+<DateField />
```

A field can never open a picker, on desktop or mobile, and it also drops the view components from your bundle. The related rendering bug (empty mobile dialog with the prop set) was fixed by #13212 in [v7.5.1](https://github.com/mui/mui-x/releases/tag/v7.5.1). I will close this issue as superseded by the deprecation.
